### PR TITLE
Improved pan behavior on groups 

### LINKED ISSUE #93: Add pan behavior to group nodes

### PR DESCRIPTION
**Pull request by JoXW100**
Fixes #93 
User can now pan by interacting with the canvas through node group backgrounds. the header of node groups is still contractible.

### CODE REVIEW

**Review by thomssonkarl — APPROVED** [review]

File: src/components/diagram/diagram.js [hunk 1]
```diff
@@ -26,14 +26,27 @@ const Diagram = () => {
   useEffect(() => {
     data.diagram?.buildDiagram(data.buildOptions)
       .then((res) => res && setState({ ...state, data: res, loading: false }))
-      .catch(abort)
+      .catch(() => {
+        console.error('failed loading diagram, retrying...')
+        window.localStorage.removeItem('diagram')
+        router.reload()
```
Comment (Caw3) [inline src/components/diagram/diagram.js:32]:
```suggestion
        router.push(".")
```

Comment (Caw3) [reply src/components/diagram/diagram.js:32]: Redirect to Dashboard instead of loading example.json

Comment (thomssonkarl) [reply src/components/diagram/diagram.js:32]: Maybe best to let the dashboard be merged before this PR.

Comment (Caw3) [reply src/components/diagram/diagram.js:32]: Doesn't matter, will work anyway.

**Review by Caw3 — APPROVED** [review]

**Pull request merged**